pkg/collector/pod_collector.go

```diff
@@ -85,32 +86,20 @@ func (c *PodCollector) GetMetrics() ([]CollectedMetric, error) {
 		return nil, err
 	}
 
-	values := make([]CollectedMetric, 0, len(pods.Items))
-
-	// TODO: get metrics in parallel
+	ch := make(chan CollectedMetric)
+	err_ch := make(chan string)
```

Comment:
I think this should be an `error` instead of `string`. As a small nitpick the variable name should be something like `errCh` to follow the go naming convention.

Reply:
Will update the variable name!

I might be missing something, but what would the value be in returning the `error`? We need the `error` AND the pod information (Name + Namespace) to log an informative error message, but after that, the `error` isn't used anymore. If do return the `error` I guess I can just drop it in the `select`?.

Side note, this implies I'd have to log the error message inside to go routine (where the pod information is available) and I'm not sure if the logging library will correctly handle concurrent logging without messing up the format?

Anyways, happy to make whichever method is preferred here!

Reply:
I simply mean it would be more semantically right to send an error type over the channel as opposed to a string. The error could still be constructed from `fmt.Errorf` and include all the relevant information.

You can also log directly from the go routine, this is not a problem. I understood that you didn't only use it for synchronizing the logging but also as a way to know what go routines where done or not.

Reply:
Ah yes, I understand now! I thought you were referencing the `error` returned from `GetMetric()`. Will update.